ros2node/ros2node/api/__init__.py

```diff
@@ -47,6 +49,32 @@ def parse_node_name(node_name):
     return NodeName(node_basename, namespace, full_node_name)
 
 
+def annotate_duplicate_nodes(node_names: List[NodeName]) -> Tuple[bool, List[str]]:
+    """
+    Annotate a list of node names with a warnings if names are not unique.
+
+    :param node_names: list of all NodeNames observed in the graph
+    :return Tuple containing:
+        bool of whether any duplicate nodes were found
+        list of all names as strings, with or without annotations, for printing
+    """
+    annotation_string = ' (non-unique name)'
+
+    any_duplicates = False
+    name_count = {}
+    for n in node_names:
+        count = name_count.get(n.full_name, 0) + 1
+        if count > 1:
+            any_duplicates = True
+        name_count[n.full_name] = name_count.get(n.full_name, 0) + 1
+
+    sorted_names = sorted(
```

Comment:
Would it not be better to return a list in the same order as the input list, i.e. do not sort it?

Reply:
you're right, it's better that way